### PR TITLE
[.Net] Add AutoGen.OpenAI package that uses OpenAI v2 SDK

### LINKED ISSUE #3404: [.Net][Feature Request]: Deprecate GPTAgent
### Is your feature request related to a problem? Please describe.

`GPTAgent` is merely `OpenAIChatAgent` plus `OpenAIMessageConnector`

### Describe the solution you'd like

Deprecate `GPTAgent` from AutoGen.OpenAI

## Migration guide

use `OpenAIChatAgent` plus message connector to replace `GPTAgent`

```csharp
// before
var agent = new GPTAgent(...)

// after
var agent = new OpenAIChatAgent(...)
    .RegisterMessageConnector()
```

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
## Related issue number
#3193

also fix #3404 

## DONT MERGE
waiting for sk to release a package working with openai beta10 package

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by joslat — APPROVED** [review]
I like it, more clean and lean. The AzureOpenAIClient is neat and better than the previous code where you had to do a configuration and "hack" it into each agent.

**Comment by LittleLittleCloud** [comment]
> I like it, more clean and lean. The AzureOpenAIClient is neat and better than the previous code where you had to do a configuration and "hack" it into each agent.

100% same here,, the `LLMConfig` is too pythonic and not quite a good fit for .NET coding style

File: dotnet/sample/AutoGen.BasicSamples/Example05_Dalle_And_GPT4V.cs [hunk 1]
```diff
@@ -74,8 +71,7 @@ public static async Task RunAsync()
                 { nameof(GenerateImage), instance.GenerateImageWrapper },
             });
         var dalleAgent = new OpenAIChatAgent(
-            openAIClient: openAIClient,
-            modelName: "gpt-3.5-turbo",
+            chatClient: openAIClient.GetChatClient("gpt-4o-mini"),
```
Comment (DavidLuong98) [inline dotnet/sample/AutoGen.BasicSamples/Example05_Dalle_And_GPT4V.cs:74]: Should we delete `gpt35Config` and `gpt4vConfig`

File: dotnet/sample/AutoGen.BasicSamples/Program.cs [hunk 2]
```diff
@@ -17,7 +17,7 @@
 allSamples.Add(new Tuple<string, Func<Task>>("User Proxy Agent", async () => { await Example06_UserProxyAgent.RunAsync(); }));
 allSamples.Add(new Tuple<string, Func<Task>>("Dynamic Group Chat - Calculate Fibonacci", async () => { await Example07_Dynamic_GroupChat_Calculate_Fibonacci.RunAsync(); }));
 allSamples.Add(new Tuple<string, Func<Task>>("LM Studio", async () => { await Example08_LMStudio.RunAsync(); }));
-allSamples.Add(new Tuple<string, Func<Task>>("LM Studio - Function Call", async () => { await Example09_LMStudio_FunctionCall.RunAsync(); }));
+//allSamples.Add(new Tuple<string, Func<Task>>("LM Studio - Function Call", async () => { await Example09_LMStudio_FunctionCall.RunAsync(); }));
```
Comment (DavidLuong98) [inline dotnet/sample/AutoGen.BasicSamples/Program.cs:20]: Should we delete since `Example09_LMStudio_FunctionCall` does not exist?

**Review by DavidLuong98 — APPROVED** [review]

**Pull request merged**